analytics.js

```diff
@@ -310,7 +322,7 @@ var Analytics = {
             }
             
             // presumably we're organizing these by date
-            if (result.data[0].date) {
+            if ((result.data.length > 0) && (result.data[0].date)) {
```

Comment:
During development, I accidentally briefly caused the new threshold feature to exclude all rows, making an empty result set. This led to a few crashes, so I added the above checks to make sure we gracefully handle that (unlikely, but possible) scenario.